lib/iris/_merge.py

```diff
@@ -1285,15 +1280,19 @@ def register(self, cube, error_on_mismatch=False):
             this :class:`ProtoCube`.
 
         """
-        signature = self._cube_signature
+        cube_signature = self._cube_signature
         other = self._build_signature(cube)
-        match = signature.match(other, error_on_mismatch)
+        match = cube_signature.match(other, error_on_mismatch)
         if match:
-            if signature.fill_value is None and other.fill_value is not None:
-                # Perform proto-cube fill-value promotion.
-                fv = other.fill_value
-                self._cube_signature = self._build_signature(self._source,
-                                                             fill_value=fv)
+            # Determine whether the fill value requires to be demoted
+            # to the default value.
+            if cube_signature.fill_value is not None:
+                if other.fill_value is None or \
+                        cube_signature.fill_value != other.fill_value:
+                    # Demote the fill value to the default.
+                    signature = self._build_signature(self._source,
```

Comment:
Why do we need a call to `_build_signature` here, but in the concatenate case you can just set `fill_value` to None?